src/ngraph/runtime/cpu/builder/quantized_conv.cpp

```diff
@@ -212,45 +214,58 @@ namespace ngraph
+                        if (ctx->first_iteration)
+                        {
+                            vector<float> dyn_scales;
+                            vector<float> dyn_post_op_scales;
+                            dyn_scales.assign(static_cast<float*>(arg4_tensor),
+                                              static_cast<float*>(arg4_tensor) + scales_size);
+                            dyn_post_op_scales.assign(static_cast<float*>(arg5_tensor),
+                                                      static_cast<float*>(arg5_tensor) +
+                                                          sum_scales_size);
+                            auto old_pops = conv_attr.get_post_ops();
+                            mkldnn::post_ops new_pops;
+                            for (int i = 0; i < old_pops.len(); i++)
                             {
-                                vector<float> dyn_scales;
-                                vector<float> dyn_post_op_scales;
-                                dyn_scales.assign(static_cast<float*>(arg4_tensor),
-                                                  static_cast<float*>(arg4_tensor) + scales_size);
-                                dyn_post_op_scales.assign(static_cast<float*>(arg5_tensor),
-                                                          static_cast<float*>(arg5_tensor) +
-                                                              sum_scales_size);
-                                auto old_pops = conv_attr.get_post_ops();
-                                mkldnn::post_ops new_pops;
-                                for (int i = 0; i < old_pops.len(); i++)
+                                if (old_pops.kind(i) == mkldnn::primitive::kind::eltwise)
                                 {
-                                    if (old_pops.kind(i) == mkldnn::primitive::kind::eltwise)
-                                    {
-                                        mkldnn::algorithm alg;
-                                        float scale, alpha, beta;
-                                        old_pops.get_params_eltwise(i, scale, alg, alpha, beta);
-                                        new_pops.append_eltwise(scale, alg, alpha, beta);
-                                    }
-                                    if (old_pops.kind(i) == mkldnn::primitive::kind::sum)
-                                    {
-                                        new_pops.append_sum(dyn_post_op_scales[0]);
-                                    }
+                                    mkldnn::algorithm alg;
+                                    float scale, alpha, beta;
+                                    old_pops.get_params_eltwise(i, scale, alg, alpha, beta);
+                                    new_pops.append_eltwise(scale, alg, alpha, beta);
+                                }
+                                if (old_pops.kind(i) == mkldnn::primitive::kind::sum)
```

Comment:
For the sake of clarity, would it make sense to either replace this `if` with `else if`, or perhaps use a `switch ... case` statement?